### DISCUSSION TITLE
Include in relations generating multiple queries instead of one join query

### DISCUSSION THREAD

**Discussion by nivruth-naka** [loopbackio/loopback-next#9344] [Q&A]
```json
{
  "where": {
    "and": [
      {
        "effectiveDate": {
          "lte": "2023-03-01"
        }
      },
      {
        "expiryDate": {
          "gte": "2023-03-01"
        }
      }
    ]
  },
  "order": "code asc",
  "include": [
    {
      "relation": "trans"
    }
  ]
}
```

I am trying to use the above filter in my [Loopback project ](https://github.com/nivruth-naka/lb-include/blob/main/src/controllers/lb-include.controller.ts) and noticing performance issues. Upon enabling [debug logging](https://github.com/nivruth-naka/lb-include/blob/main/lb-include.log) for the loopback connector I noticed that the include relation is being queried separately sequentially after a separate main query is executed. My understanding from the docs was that it would be a single join query but instead I am seeing 11 queries (1 main query which returns 10 rows and 10 sequential queries for each result from the main query for the include relation part). Is this a bug? This is a one-one association and in my [models](https://github.com/nivruth-naka/lb-include/tree/main/src/models) I am using [@hasOne](https://github.com/nivruth-naka/lb-include/blob/main/src/models/main.model.ts#L70) and [@belongsTo](https://github.com/nivruth-naka/lb-include/blob/main/src/models/trans.model.ts#L42) as well. Please assist.

**Comment by shubhamp-sf** [comment]
For anyone stopping by here in future, the discussion on this is moved here: https://github.com/loopbackio/loopback-next/issues/9345
